### ISSUE TITLE
raw definition of XCM AccountId hampers UX in js/apps

### ISSUE THREAD

**Issue by brenzi** [paritytech/polkadot#4187]
The choice of `[u8; 32]` here doesn't tell js/apps that it should provide the account selector
https://github.com/paritytech/polkadot/blob/cc24fb87b702db6510d08c689e442b6be384d798/xcm/src/v0/junction.rs#L120

causes:
https://github.com/polkadot-js/apps/issues/6455

could you use `AccountId` instead? I see the naming ambiguity with the outer type, but somehow this should be resolved

**Comment by bkchr** [comment]
That is something that needs to be improved on on the UI side.

**Comment by xlc** [comment]
For AccountId type with 32 bytes length, it makes total sense to use `AccountId32` for it?

**Comment by bkchr** [comment]
Did you read @jacogr comment?

> I do believe the upstream definition is correct - it can be a non-AccountId on the recipient chain, so it doesn't map to an AccountId32, i.e. it doesn't make assumptions on what the destination chain has as a type. (Same with the [u8; 20] type)

**Comment by xlc** [comment]
Yes and I don't agree with it. This junction is called `AccountId32` and the `AccountId32` type on Substrate is `An opaque 32-byte cryptographic identifier.` I can't think a case where this will cause any issues.

https://github.com/paritytech/substrate/blob/0187df5d4562f714956983384387ddccb2117140/primitives/core/src/crypto.rs#L467

**Comment by bkchr** [comment]
Okay, fine.
